test/e2e/lib/hooks/mocha.js

```diff
@@ -2,28 +2,70 @@
  * External dependencies
  */
 import config from 'config';
+import path from 'path';
+import { mkdtemp } from 'fs/promises';
 
 /**
  * Internal dependencies
  */
-import { buildHooks as buildVideoHooks } from './video';
+import { buildHooks as buildVideoHooks, isVideoEnabled } from './video';
 import { buildHooks as buildBrowserHooks } from './browser';
 
 const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
 
-export const mochaHooks = () => {
-	const videoHooks = buildVideoHooks();
+export const mochaHooks = async () => {
+	const tempDir = await mkdtemp( path.resolve( __dirname, '../../test-' ) );
+	let driver;
+	const hooks = {
+		afterAll: [],
+		beforeAll: [],
+		afterEach: [],
+	};
+
+	if ( isVideoEnabled() ) {
+		const {
+			startFramebuffer,
+			stopFramebuffer,
+			takeScreenshot,
+			startVideoRecording,
+			saveVideoRecording,
+			stopVideoRecording,
+		} = buildVideoHooks();
+
+		hooks.beforeAll.push( async function () {
+			await startFramebuffer();
+			await startVideoRecording( { tempDir } );
+		} );
+
+		hooks.afterEach.push( async function () {
+			if ( this.currentTest && this.currentTest.state === 'failed' ) {
```

Comment:
TIL: We can't use `this.currentTest?.state` because it is not supported by `esm`